packages/next-codemod/transforms/new-link.ts

```diff
@@ -86,13 +86,24 @@ export default function transformer(file: FileInfo, api: API) {
         const hasProps = props.length > 0
 
         if (hasProps) {
-          // Add props to <Link>
-          $link.get('attributes').value.push(...props)
+          // Add only unique props to <Link> (skip duplicate props)
```

Comment:
I wonder if we can move this to some sort of a global helper somewhere? Seems like a nice common pattern we might want to share––just a thought. Thanks for the addition and solving my issue #41925 !

Reply:
Aw, missed our chance to merge maybe. @styfle think it's worth sharing this logic around the other codemods? Think we should be concerned we'll see this again with other mods?

Great stuff either way, thanks for addressing 💯

Reply:
@andrewmartin Which other codemods would share it?

Reply:
@styfle I'm just speculating, and I am being admittedly only inquisitive (not very thorough), but I did see this:

https://github.com/vercel/next.js/blob/3f628006d23b41e6adfaa2b810fa448358eb729b/packages/next-codemod/transforms/next-image-experimental.ts#L12

So, wondering if, in theory, someone had put a `style` attribute on an existing JSX element here, would it be duplicated? Sorry for not testing myself, just throwing the idea out there first.

Reply:
Furthermore, as more codemods will be added, it'd be nice to have this logic compartmentalized. I'd imagine that any new codemod that modifies an existing element's props would want to have similar logic and avoid duping.